File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;
class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int answer = 0;
        int n = queue1.length;
        
        Queue<Integer> q1 = new ArrayDeque<Integer>();
        Queue<Integer> q2 = new ArrayDeque<Integer>();
        
        long q1Sum = 0;
        long q2Sum = 0;
        
        for (int value: queue1){
            q1.add(value);
            q1Sum+=value;
        }
        for (int value: queue2){
            q2.add(value);
            q2Sum+=value;
        }
        
        
        for(int i=0; i<n*4; i++){
            if(q1Sum>q2Sum){
                int temp = q1.poll();
                q2.add(temp);
                q1Sum-=temp;
                q2Sum+=temp;
                answer++;
            }else if(q1Sum<q2Sum){
                int temp = q2.poll();
                q1.add(temp);
                q2Sum-=temp;
                q1Sum+=temp;
                answer++;                
            }else{
                return answer;
            }
        }
        
        return -1;
    }
}

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.js
function solution(queue1, queue2) {
    var answer = 0;
    const qLength = queue1.length;
    
    let q1Index=0;
    let q2Index=0;
    
    let q1Sum = queue1.reduce((a, b)=>a+b);
    let q2Sum = queue2.reduce((a, b)=>a+b);
    
    for(let i=0; i<qLength*4; i++){
        if(queue1.length-1<q1Index||queue2.length-1<q2Index) {
            return -1;
        }
        
        
        if(q1Sum>q2Sum){
            queue2.push(queue1[q1Index]);
            
            q1Sum-=queue1[q1Index];
            q2Sum+=queue1[q1Index];
            
            q1Index++;
            
            answer++;
        }else if(q2Sum>q1Sum){
            queue1.push(queue2[q2Index]);
            
            q1Sum+=queue2[q2Index];
            q2Sum-=queue2[q2Index];
            
            q2Index++;
            
            answer++;
        }else{
            return answer;
        }
        // if()
    }
    
    return -1;
}